lib/usdUfe/ufe/UsdHierarchy.cpp

```diff
@@ -257,6 +257,19 @@ UsdHierarchy::insertChildCmd(const Ufe::SceneItem::Ptr& child, const Ufe::SceneI
     return UsdUndoInsertChildCommand::create(fItem, downcast(child), downcast(pos));
 }
 
+Ufe::InsertChildCommand::Ptr
+UsdHierarchy::appendChildVerifyRestrictionsCmd(const Ufe::SceneItem::Ptr& child)
+{
+    const auto childCmd = appendChildCmd(child);
```

Comment:
A better and cleaner solution would be to create a new cmd that inherits from `UsdUndoInsertChildCommand` and in the new constructor check the restrictions and throw the exception.